### ISSUE TITLE
Add compatibility/switch to openshift-restclient 0.9+

### ISSUE THREAD

**Issue by mmazur** [kubevirt/ansible-kubevirt-modules#189]
/kind enhancement

Upstream openshift client is currently at versions 0.8.x. However, the master branch is getting ready for 0.9, which will be noticeably incompatible with current code. Once that gets released, we'll need to make a decision wrt porting, etc.

**Comment by mmazur** [comment]
I've talked to upstream. Turns out we should look into this asap.

0.9 release is not purposefully backwards incompatible. There is some stuff that's being dropped, like the old way of doing watchers, but that's because in 0.8 it's possible to do it in the new, future–proof way. Potentially there are also other things like that. Plus upstream said that if we have some issues with 0.9, we can ping him about it, as he's willing to consider fixing it upstream as not to break existing code.

So, tl;dr – if we take care of this now (make our current codebase work against openshift-client's master branch), there's a good chance that we can make our existing code work both with 0.8 and with 0.9 once that comes out. If we wait until 0.9 is released to deal with this issue, we're setting ourselves up for problematic incompatibilities.

CC @pkliczewski @machacekondra

**Comment by mmazur** [comment]
Oh, and for reference: 

> fabianvf | in both the 0.8 and 0.9 branches we added a watach function to the client itself: https://github.com/openshift/openshift-restclient-python/blob/release-0.8/openshift/dynamic/client.py#L266

**Comment by machacekondra** [comment]
If we just start to use watch from Kubernetes client it should be enough no?

**Comment by mmazur** [comment]
It's a start, but iirc something also changed wrt to exception handling. It was a few weeks ago, I don't remember the details. Bottom line is we need to make sure this gets taken care of during the next sprint at most.

**Comment by mmazur** [comment]
I've bumped the prio, since this is especially important to get done and merged into upstream before they do their freeze.

**Comment by pkliczewski** [comment]
@mmazur Do we want to still work on this item?

**Comment by mmazur** [comment]
This has been completed, closing.
